test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,644 @@
+        super.setup();
+
+        // 1. Start ZK
+        this.zkTestServer = new TestingServerStarter().start();
+
+        // 2. Start Pravega SegmentStore service.
+        this.serviceBuilder = ServiceBuilder.newInMemoryBuilder(ServiceBuilderConfig.getDefaultConfig());
+        serviceBuilder.initialize();
+        this.tableStore = serviceBuilder.createTableStoreService();
+        this.streamSegmentStore = serviceBuilder.createStreamSegmentService();
+
+        this.serverListener = new PravegaConnectionListener(false, servicePort, this.streamSegmentStore,
+                this.tableStore, executorService());
+        this.serverListener.startListening();
+
+        // 3. Start Pravega Controller service
+        this.controllerWrapper = new ControllerWrapper(zkTestServer.getConnectString(), false,
+                controllerPort, serviceHost, servicePort, containerCount);
+        this.controllerWrapper.awaitRunning();
+        this.controller = controllerWrapper.getController();
+
+        //4. Create Scope
+        this.controller.createScope(SCOPE).get();
+        ClientConfig clientConfig = ClientConfig.builder().build();
+        SocketConnectionFactoryImpl connectionFactory = new SocketConnectionFactoryImpl(clientConfig);
+        this.connectionPool = new ConnectionPoolImpl(clientConfig, connectionFactory);
+        this.keyValueTableFactory = new KeyValueTableFactoryImpl(SCOPE, this.controller, this.connectionPool);
+
+        //5. setting up controller service instance
+        this.controllerService = new ControllerServiceImpl(controllerWrapper.getControllerService(),
+                GrpcAuthHelper.getDisabledAuthHelper(), requestTracker, true, 2);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        this.controller.close();
+        this.connectionPool.close();
+        this.controllerWrapper.close();
+        this.serverListener.close();
+        this.serviceBuilder.close();
+        this.zkTestServer.close();
+        this.keyValueTableFactory.close();
+    }
+
+    /**
+     * Create Stream and KVTable with same name under same scope.
+     */
+    @Test
+    public void testCreateStreamKVTable() throws ExecutionException, InterruptedException {
+        String newScope = "NewScope";
+        KeyValueTableInfo newKVT = newKeyValueTableName(newScope);
+        String streamName = newKVT.getKeyValueTableName();
+        int streamSegments = 1;
+        String readerGroupName = "readerGroup";
+        String readerName = "reader";
+        String routingKey = "routingKey";
+        String streamEvent = "Stream event";
+        int numEvents = 100;
+
+        final ScalingPolicy scalingPolicy = ScalingPolicy.fixed(streamSegments);
+        final StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(scalingPolicy)
+                .build();
+
+        assertTrue("The requested Scope must get created", this.controller.createScope(newScope).join());
```

Comment:
You could use the utility method in `TestUtils` to create scope and stream: 

https://github.com/pravega/pravega/blob/ede135d5cc72aceb2add3ab6d6cef2f1ac8238fa/test/integration/src/main/java/io/pravega/test/integration/utils/TestUtils.java#L56